src/modules/flow/iio/nodes.c

```diff
@@ -106,14 +238,16 @@ create_device_channels(struct gyroscope_data *mdata, int device_id)
 static int
 gyroscope_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
 {
-    struct gyroscope_data *mdata = data;
+    struct iiodevice_data *mdata = data;
     const struct sol_flow_node_type_iio_gyroscope_options *opts;
     int device_id;
 
     SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options, SOL_FLOW_NODE_TYPE_IIO_GYROSCOPE_OPTIONS_API_VERSION,
         -EINVAL);
     opts = (const struct sol_flow_node_type_iio_gyroscope_options *)options;
 
+    mdata->type = IIO_GYROSCOPE;
```

Comment:
this is not an instance member, but a class one. All gyroscopes share that, thus add this to `extra_methods` as you did for the out_port.